module/zfs/zil.c

```diff
@@ -1461,8 +1461,7 @@ zil_lwb_commit(zilog_t *zilog, itx_t *itx, lwb_t *lwb)
 				lrw->lr_offset += dnow;
 				lrw->lr_length -= dnow;
 				ZIL_STAT_BUMP(zil_itx_needcopy_count);
-				ZIL_STAT_INCR(zil_itx_needcopy_bytes,
-				    lrw->lr_length);
+				ZIL_STAT_INCR(zil_itx_needcopy_bytes, dnow);
```

Comment:
This same change should be made when incrementing the `zil_itx_copied_bytes` above and `zil_itx_indirect_bytes` below.  They can't underflow in the same way but `dnow` is the accurate length to use.

Reply:
I think the simple change is better than where we are now (dnow? :-)

I'd like to consider recording both dnow and lr_length, but that should be a separate PR, if and when it proves useful. And I'd like to move any test case to the separate PR.

Reply:
@chrisrd good point!  OK, then this LGTM and nice job reproducing the issue to veirfy the fix.